packages/web-components/src/components/rux-select/test/select.spec.ts

```diff
@@ -74,7 +74,88 @@ test.describe('Select', () => {
         isFocused = await el.evaluate((el) => el === document.activeElement)
         expect(isFocused).toBeTruthy()
     })
+    test('Options that are dynamically added can be synced to select', async ({
+        page,
+    }) => {
+        const template = `
+          <rux-select value="flash">
+          <rux-option value="" label="Select"></rux-option>
+          <rux-option-group label="Group">
+              <rux-option value="flash" label="Flash"></rux-option>
+          </rux-option-group>
+        </rux-select>
+        <rux-button>Add to options</rux-button>
+        `
+        await page.setContent(template)
+        await page.addScriptTag({
+            content: `
+              const sel = document.querySelector('rux-select')
+              const optGroup = document.querySelector('rux-option-group')
+              const init = [
+                  { value: 'aquaman', name: 'Aquaman' },
+                  { value: 'batman', name: 'Batman' },
+              ]
+              const btn = document.querySelector('rux-button')
+              btn.addEventListener('click', () => {
+                  init.map((hero) => {
+                      let newOption = document.createElement('rux-option')
+                      newOption.label = hero.name
+                      newOption.value = hero.value
+                      optGroup.appendChild(newOption)
+                  })
+        })
+        `,
+        })
+        //selected value should be flash
+        const sel = page.locator('rux-select').locator('select')
```

Comment:
you don't have to chain locators here. you can just do `page.locator('rux-select select')` and playwright will reach in the shadow dom.

Reply:
Good to know!